draco/synthesis/noise.py

```diff
@@ -53,6 +53,67 @@ def process(self, data):
         return data
 
 
+class GaussianNoise(task.SingleTask):
+    """Add Gaussian Noise from Receiver Temperature."""
+
+    recv_temp = config.Property(proptype=float, default=50.0)
+    ndays = config.Property(proptype=float, default=733.)
+    seed = config.Property(proptype=int, default=None)
+    set_weights = config.Property(proptype=bool, default=True)
+
+    def process(self, data):
+
+
+        data.redistribute('freq')
+
+        visdata = data.vis[:]
+
+        # Get the time interval
+        if isinstance(data, containers.SiderealStream):
+            dt = 240 * (data.ra[1] - data.ra[0]) * STELLAR_S
+            ntime = len(data.ra)
+        else:
+            dt = data.time[1] - data.time[0]
+            ntime = len(data.time)
+
+        df = data.index_map['freq']['width'][0] * 1e6
+        nfreq = data.vis.local_shape[0]
+        nprod = len(data.index_map['prod'])
+
+        #with mpi_random_seed(self.seed):
+
+        # Calculate the number of samples
+        nsamp = int(self.ndays * dt * df)
+        std = self.recv_temp / np.sqrt(2 * nsamp)
+        ndraws = nfreq * ntime * nprod
+
+        std_real = std * np.random.standard_normal(ndraws).reshape(nfreq, nprod, ntime)
```

Comment:
These variable names are confusing. `std_` usually implies it's a standard deviation, there are actual realisations so call them something more appropriate.